private/buf/buffetch/ref_parser.go

```diff
@@ -309,62 +323,80 @@ func (a *refParser) checkDeprecated(parsedRef internal.ParsedRef) {
 	}
 }
 
-func processRawRef(rawRef *internal.RawRef) error {
-	// if format option is not set and path is "-", default to bin
-	var format string
-	var compressionType internal.CompressionType
-	if rawRef.Path == "-" || app.IsDevNull(rawRef.Path) || app.IsDevStdin(rawRef.Path) || app.IsDevStdout(rawRef.Path) {
-		format = formatBin
-	} else {
-		switch filepath.Ext(rawRef.Path) {
-		case ".bin":
+type rawRefProcessorConfig struct {
+	allowProtoFileRef bool
+}
+
+func newRawRefProcessor(config *rawRefProcessorConfig) func(*internal.RawRef) error {
```

Comment:
I appreciate the configuration struct, but we can probably get away with a simple `bool` for now (similar to the comment above).